### PR TITLE
Clear the DartdocEntry status flags on regression.

### PR DESCRIPTION
**Pull request by isoos**
Handles the first part of the core issue in #1786.

### CODE REVIEW

File: app/lib/dartdoc/dartdoc_runner.dart [hunk 1]
```diff
@@ -192,6 +192,10 @@ class DartdocJobProcessor extends JobProcessor {
           job.packageName, job.packageVersion);
       if (entry.isRegression(oldEntry)) {
         _logger.severe('Regression detected in $job, aborting upload.');
+        if (oldEntry.isLatest != entry.isLatest ||
+            oldEntry.isObsolete != entry.isObsolete) {
+          await dartdocBackend.updateOldEntry(oldEntry, entry);
```
Comment (jonasfj) [inline app/lib/dartdoc/dartdoc_runner.dart]: So, basically, if `isLatest` or `isObsolete` have changed, you still want to update the old entry, even if the job failed?

Please maybe add a one-line comment stating that..

**Comment by isoos** [comment]
> Isn't the underlying problem here that we have duplicate information in the DartDocEntry, since isLatest and isObsolete can be computed from other places, right?

Correct. Another option is to use an extra Datastore check (+memcache) on each content that is being served, and then we can sunset the flag too. Which would you prefer?

**Comment by jonasfj** [comment]
I have no opinion at this point.. :)

**Comment by isoos** [comment]
I'm merging this with the doc now. We can easily remove this if we want to change the frontend serving part to not use the fields.

**Pull request merged**